pkg/controller/controller.go

```diff
@@ -705,10 +705,16 @@ func (s *Service) DeleteVolume(ctx context.Context, req *csi.DeleteVolumeRequest
 // ControllerPublishVolume prepares Volume/FileSystem to be consumed by node by attaching/allowing access to the host.
 func (s *Service) ControllerPublishVolume(ctx context.Context, req *csi.ControllerPublishVolumeRequest) (*csi.ControllerPublishVolumeResponse, error) {
 	id := req.GetVolumeId()
+	nodeID := req.GetNodeId()
+
 	if id == "" {
 		return nil, status.Error(codes.InvalidArgument, "volume ID is required")
 	}
 
+	if nodeID == "" {
+		return nil, status.Error(codes.InvalidArgument, "Node ID is required")
+	}
```

Comment:
I noticed that `nodeId` validation already exists later in the method https://github.com/dell/csi-powerstore/blob/5df3c12e3adaf14459e75e25a5d25a18ba5b6201/pkg/controller/controller.go#L749-L752 

If the intent is to fail fast, we could consider relocating the existing check instead of duplicating it.

Reply:
Agree with @AkshaySainiDell